server/lib/nl/detection/place.py

```diff
@@ -279,6 +244,66 @@ def _get_non_place_entities(all_entities: List[str],
   return entities
 
 
+#
+# Helper function to get a list of query parts for a query.
+#
+def _get_query_parts(query: str, debug_logs: Dict[str, any],
+                     for_place: bool) -> List[QueryPart]:
+  if for_place:
+    query_items = dc.recognize_places(query)
+    items_key = "places"
+  else:
+    query_items = dc.recognize_entities(query)
+    items_key = "entities"
+
+  query_parts = []
+  for item in query_items:
+    if 'span' not in item:
+      continue
+    if items_key in item and item[items_key] and 'dcid' in item[items_key][0]:
+      dcids = [d['dcid'] for d in item[items_key] if 'dcid' in d]
+      query_parts.append(QueryPart(substr=item['span'].lower(), dcids=dcids))
+
+      # log the dcids:
+      debug_logs[item['span']] = dcids
+    else:
+      query_parts.append(QueryPart(substr=item['span'].lower()))
+  return query_parts
+
+
+#
+# Helper function to get the list of strings that correspond to a resolved dcid
+# and the query string with all those strings stripped out
+#
+def _get_stripped_string(query_parts: List[QueryPart],
+                         resolved_dcids: Set[str]) -> tuple[List[str], str]:
+  resolved_dcids_str = []
+  stripped_query_parts = []
+  for p in query_parts:
+    if any([p_dcid in resolved_dcids for p_dcid in p.dcids]):
+      # if any of the dcids in the query part is resolved, consider the query part
+      # resolved and add the original string to resolved_things_str
+      resolved_dcids_str.append(p.substr)
+    else:
+      stripped_query_parts.append(p.substr)
+  stripped_query_str = ' '.join(stripped_query_parts)
+  return resolved_dcids_str, stripped_query_str
+
+
+#
+# Gets the resolved entities from a list of query parts with non place entities
+# recognized.
+#
+def _get_resolved_entities(query_parts: List[QueryPart]) -> List[Entity]:
+  entity_dcids = set()
+  for e in query_parts:
+    if not e.dcids:
+      continue
+    entity_dcids.update(e.dcids)
+
+  return _get_non_place_entities(list(entity_dcids), [])
```

Comment:
Why is 2nd arg empty? Comment...

Reply:
made the second arg optional in the helper function and updated the helper function definition